src/cmap/auth/gssapi.ts

```diff
@@ -30,60 +51,161 @@ export class GSSAPI extends AuthProvider {
       mechanismProperties['gssapiServiceName'] ||
       'mongodb';
 
-    const MongoAuthProcess = Kerberos.processes.MongoAuthProcess;
+    function externalCommand(
+      command: object,
+      cb: (
+        err: Error | MongoError | undefined,
+        result: { result: { payload: string; conversationId: any } }
+      ) => void
+    ) {
+      return connection.command('$external.$cmd', command, cb);
+    }
+
+    initialize(
+      {
+        retries: 10,
+        host,
+        port,
+        username,
+        password,
+        serviceName: gssapiServiceName,
+        canonicalizeHostName:
+          typeof mechanismProperties.gssapiCanonicalizeHostName === 'boolean'
+            ? mechanismProperties.gssapiCanonicalizeHostName
+            : false
+      },
+      (err, context) => {
+        if (err) return callback(err);
 
-    const authProcess = new MongoAuthProcess(host, port, gssapiServiceName, mechanismProperties);
+        context!.client.step('', (err, payload) => {
```

Comment:
The stricter ruleset we've been using will not permit these explicit casting statements, so let's just start off by not using them. Can you check if `context` is nullish and bail early with an error, otherwise destructure the `client` from it.